piecewise/regressor.py

```diff
@@ -235,53 +285,115 @@ def _get_initial_segments(t, v):
     still be suboptimal initializations, as in this case, where the two 1s will
     be initialized in the same segment: [19, 10, 1, 1, -8, -17]
     """
+
+    # creates segments from an array of start, end indices
+    def _init_data(ranges):
```

Comment:
Can this function be named more descriptively, like `_build_segments()` or something along those lines?

Reply:
Done